R/secchi_workflow.R

```diff
@@ -0,0 +1,189 @@
+    return(result)
+  }
+  retry = 1
+  while (retry < retries){
+    result = safeWQP(...)
+    if (!is.null(result)){
+      retry = retries
+    } else {
+      message('query failed, retrying')
+      retry = retry+1
+    }
+  }
+  return(result)
+}
+
+library(dplyr)
+library(dataRetrieval)
+library(lubridate)
+library(ggplot2)
+
+characteristicNames = c("Depth, Secchi disk depth", "Depth, Secchi disk depth (choice list)", "Secchi Reading Condition (choice list)", "Secchi depth", "Water transparency, Secchi disc")
+siteTypes = "Lake, Reservoir, Impoundment"
+
+lower.48 <- stateCd$STATE[1:50]
+lower.48 <- lower.48[!(stateCd$STUSAB[1:50] %in% c("AL","HI"))]
+# To get the full data set used to produce the figure in the text:
+# get_us_secchi(outfile="all_secchi_usa.rds",
+#                   characteristicNames,
+#                   siteTypes,
+#                   stateCode = lower.48,
+#                   '1980-01-01', '2016-01-01')
+
+# To get the small subset to test the workflow:
+get_us_secchi(outfile="sub_secchi_WI_MN.rds",
+                characteristicNames,
+                siteTypes,
+                stateCode = c("01","12","27","55"), #AL, FL, WI, and MN
+                '2000-01-01', '2016-01-01')
+
+infile <- "sub_secchi_WI_MN.rds"
+secchi.data <- readRDS(infile)
+
+regions <- data.frame(STATE_NAME = c('Montana', 'Wyoming', 'Idaho', 'Washington', 'Oregon', 'California', 'Nevada',
+                                     'Arizona', 'New Mexico', 'Colorado', 'Utah'), group='west', stringsAsFactors = FALSE) %>% 
+  rbind(data.frame(STATE_NAME=c('Ohio', 'Indiana', 'Illinois', 'Wisconsin', 'Missouri', 'Iowa', 'Minnesota', 'Kansas',
+                                'Nebraska', 'South Dakota', 'North Dakota', 'Michigan'), group='midwest', stringsAsFactors = FALSE)) %>% 
+  rbind(data.frame(STATE_NAME=c('Maine', 'New Hampshire', 'Vermont', 'Massachusetts', 'Rhode Island', 'Connecticut', 
+                                'New Jersey', 'Pennsylvania', 'New York'), group='northeast', stringsAsFactors = FALSE)) %>% 
+  rbind(data.frame(STATE_NAME=c('Florida', 'Georgia', 'Louisiana', 'Arkansas', 'Oklahoma', 'Texas', 'South Carolina', 
+                                'North Carolina', 'Virginia', 'Kentucky', 'Tennessee', 'West Virginia', 'Maryland', 
+                                'Delaware', 'Alabama', 'Mississippi'), group='south', stringsAsFactors = FALSE))
+
+secchi.data.1 <- left_join(secchi.data, stateCd, by=c("StateCode" = "STATE")) %>%
+  left_join(regions, by="STATE_NAME") %>%
+  mutate(week = lubridate::week(Date)) %>% 
+  filter(week > 13 & week < 47) %>%
+  group_by(week, group) %>%
+  summarize(med = median(secchi, na.rm=TRUE), 
+            q25 = quantile(secchi, na.rm=TRUE, probs = .25),
+            q75 = quantile(secchi, na.rm=TRUE, probs = .75))
+
+secchi.plat <- ggplot(data=secchi.data.1) +
+  geom_line(aes(x=week, y=med, color=group), size=2) +
+  ylab("Secchi depth (m)") +
+  xlab("Week of the year") +
+  theme_bw()
+
```

Comment:
should include the line that actually prints the fig (or saves as a .png or something)